### PR TITLE
Update API reference - Closes #554

### LINKED ISSUE #554: Update API specification
## Which content is outdated?
The Lisk SDK API specification.

## What should be the new content?
It needs to reflect the API changes introduced in the Lisk SDK 4.0

### PR DESCRIPTION
**Pull request by Tschakki**
Closes #554

### CODE REVIEW

File: modules/ROOT/pages/reference/api.adoc [hunk 1]
```diff
@@ -1113,47 +958,82 @@ __required__|Error message containing details of the error +
...
-|**secondPublicKey** +
-__optional__|The second public key is derived from the second private key of an account, if the owner activated a second passphrase for her/his account. +
+It can be used to validate that the private key belongs to the owner, but does not provide access to the owners private key. +
 **Example** : `"968ba2fa993ea9dc27ed740da0daf49eddd740dbd7cb1cb4fc5db3a20baf341b"`|string (publicKey)
+|**rewards** +
+__optional__|Total sum of block rewards that the delegate has forged. +
+**Example** : `"510000000"`|string
+|**totalVotesReceived** +
+__optional__|The total votes received by the delegate.
+Represents the total amount of Lisk (in Beddows), that the delegates voters voted this delegate. +
```
Comment (chris529) [inline modules/ROOT/pages/reference/api.adoc]:
```suggestion
Represents the total amount of Lisk (in Beddows), that the delegates voters voted for this delegate. +
```

File: modules/ROOT/pages/reference/api.adoc [hunk 2]
```diff
@@ -1325,68 +1205,19 @@ __required__|<<_meta,Meta>>
...
-The voters weight decides which rank the delegate gets in relation to the other delegates and their voters weights. +
-**Example** : `"1081560729258"`|string
+|**consecutiveMissedBlocks** +
+__optional__|Number of blocks that the delegate missed consecutively. +
+**Example** : `10`|integer
+|**isBanned** +
+__optional__|Whether the delegate is banned or not. +
+**Example** : `false`|boolean
+|**lastForgedHeight** +
+__optional__|Height of the block which delegate forged last time. +
```
Comment (chris529) [inline modules/ROOT/pages/reference/api.adoc]:
```suggestion
__optional__|Height of the block after the latest forging that was executed by the delegate.  +
```

File: modules/ROOT/pages/reference/api.adoc [hunk 3]
```diff
@@ -1325,68 +1205,19 @@ __required__|<<_meta,Meta>>
...
+**Example** : `100`|integer
+|**pomHeights** +
+__optional__|Height of blocks where delegate has been reported for misbehavior.|< integer > array
```
Comment (chris529) [inline modules/ROOT/pages/reference/api.adoc]:
```suggestion
__optional__|Height of blocks where the delegate has been reported for misbehavior.|< integer > array
```

File: modules/ROOT/pages/reference/api.adoc [hunk 4]
```diff
@@ -1407,47 +1238,14 @@ Amount of Lisk the delegate account owns. +
...
-|**votes** +
-__required__|List of placed votes by the queried account.|<<_vote,Vote>> array
-|**votesAvailable** +
-__required__|Number of votes that are available for the queried account.
-Derives from 101 (max possible votes) - votesUsed(alreadu used votes) +
-**Example** : `40`|integer
-|**votesUsed** +
-__required__|Number of votes that are already placed by the queried account. +
-**Example** : `2`|integer
+__required__|Accounts this delegate voted for.|< <<_vote,Vote>> > array
```
Comment (chris529) [inline modules/ROOT/pages/reference/api.adoc]:
```suggestion
__required__|Accounts which this delegate voted for.|< <<_vote,Vote>> > array
```

File: modules/ROOT/pages/reference/api.adoc [hunk 5]
```diff
@@ -1874,25 +1617,25 @@ __Type__ : integer (int32)
 
 [[_paramerror]]
 === ParamError
-Collection of errors on a particular parameter.
+Collection of errors in a particular parameter.
```
Comment (chris529) [inline modules/ROOT/pages/reference/api.adoc]:
```suggestion
Collection of errors in a specific parameter.
```

**Review by chris529 — COMMENTED** [review]
looks very good, just a few minor points ;-)

**Comment by Tschakki** [comment]
Hi @chris529 !
Although your suggestions seem ok, I am not applying them here, as the API spec is generated out of the `swagger.yml` file in the lisk-sdk repo. We would need to fix the issues there. But imho it are only minor issues so it can stay for now.

**Comment by chris529** [comment]
OK, cool..

**Review by chris529 — COMMENTED** [review]
Cool... :-)

File: modules/ROOT/pages/reference/api.adoc [hunk 6]
```diff
@@ -451,15 +358,15 @@ __optional__|Offset value for results|integer (int32)|`0`
...
 ....
 GET /delegates/{address}/forging_statistics
 ....
 
 
 ==== Description
-By passing an existing delegate address and the desired unix timestamps, it is possible to get its forging statistics within the specified timespan.
-If no timestamps are provided, it will use the timestamps from Lisk epoch to current date.
+By passing an existing delegate address and the desired unix timestamps, it is possible to acquire the forging statistics within the specified timespan.
+If no timestamps are provided, the timestamps from Lisk epoch to the current date will be used.
```
Comment (chris529) [inline modules/ROOT/pages/reference/api.adoc:369]:
```suggestion
If no timestamps are provided, the timestamps from the Lisk epoch to the current date will be used.
```

File: modules/ROOT/pages/reference/api.adoc [hunk 7]
```diff
@@ -1113,47 +958,82 @@ __required__|Error message containing details of the error +
...
+__required__||<<_delegate,Delegate>>
+|**isDelegate** +
+__optional__|The value indicating if the account is a deleage or not.|boolean
+|**keys** +
+__required__||<<_multisignatureasset,MultisignatureAsset>>
+|**missedBlocks** +
+__optional__|Total number of blocks that the delegate has missed. +
+**Example** : `427`|integer
+|**nonce** +
+__required__|The current nonce associated to account for transaction processing. +
```
Comment (chris529) [inline modules/ROOT/pages/reference/api.adoc:983]:
```suggestion
__required__|The current nonce associated to the account for transaction processing. +
```

File: modules/ROOT/pages/reference/api.adoc [hunk 8]
```diff
@@ -2261,29 +1925,58 @@ __required__||<<_offset,Offset>>
...
 
 
+[[_unlocking]]
+=== Unlocking
+
+[options="header", cols=".^3a,.^11a,.^4a"]
+|===
+|Name|Description|Schema
+|**amount** +
+__required__|Amount the account voted the delegate for in multiples of 10 Lisk. +
```
Comment (chris529) [inline modules/ROOT/pages/reference/api.adoc:1940]:
```suggestion
__required__|Amount  that the account voted the delegate for, in multiples of 10 Lisk. +
```

File: modules/ROOT/pages/reference/api.adoc [hunk 9]
```diff
@@ -2261,29 +1925,58 @@ __required__||<<_offset,Offset>>
...
 [[_vote]]
 === Vote
 
 [options="header", cols=".^3a,.^11a,.^4a"]
 |===
 |Name|Description|Schema
-|**address** +
-__required__|Lisk Address of the delegate the queried account voted for. +
+|**amount** +
+__required__|Amount the account voted the delegate for in multiples of 10 Lisk. +
```
Comment (chris529) [inline modules/ROOT/pages/reference/api.adoc:1958]:
```suggestion
__required__|Amount that the account voted the delegate for, in multiples of 10 Lisk. +
```

File: modules/ROOT/pages/reference/api.adoc [hunk 10]
```diff
@@ -2261,29 +1925,58 @@ __required__||<<_offset,Offset>>
...
+
+
+[[_votewithdelegateproperties]]
+=== VoteWithDelegateProperties
+
+[options="header", cols=".^3a,.^11a,.^4a"]
+|===
+|Name|Description|Schema
+|**amount** +
+__required__|Amount the account voted the delegate for in multiples of 10 Lisk. +
```
Comment (chris529) [inline modules/ROOT/pages/reference/api.adoc:1973]:
```suggestion
__required__|Amount that the account voted the delegate for, in multiples of 10 Lisk. +
```

File: modules/ROOT/pages/reference/api.adoc [hunk 11]
```diff
@@ -2296,12 +1989,11 @@ __required__|Username of the delegate the queried account voted for. +
...
 __required__|The Lisk address of the account that voted for the queried delegate. +
 **Example** : `"12668885769632475474L"`|string (address)
-|**balance** +
-__required__|Balance of the account that voted for the queried delegate. +
-**Example** : `"1081560729258"`|string
 |**publicKey** +
 __required__|Public key of the account that voted for the queried delegate. +
 **Example** : `"968ba2fa993ea9dc27ed740da0daf49eddd740dbd7cb1cb4fc5db3a20baf341b"`|string (publicKey)
+|**votes** +
+__required__|All votes the voter for the queried delegate voted for.|< <<_vote,Vote>> > array
```
Comment (chris529) [inline modules/ROOT/pages/reference/api.adoc:1996]:
```suggestion
__required__|All votes that the voter for the queried delegate voted for.|< <<_vote,Vote>> > array
```

**Review by chris529 — COMMENTED** [review]
Excellent, just a few very minor points :-)

File: modules/ROOT/pages/reference/api.adoc (hunk 11 above)
Comment (Tschakki) [reply modules/ROOT/pages/reference/api.adoc:1996]: As I already mentioned in the comment before, please don't update the content here, as it is autogenerated... I wont apply these changes here as they will e overwritten in future.

**Comment by Tschakki** [comment]
As I already mentioned in the comment before, please don't update the content here, as it is autogenerated... I wont apply these changes here as they will be overwritten in future.

**Review by chris529 — APPROVED** [review]

**Pull request merged**